src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/search/parameter/SearchRequestCustomDeserializerTest.java

```diff
@@ -0,0 +1,149 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.parameter;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.JsonTest;
+import org.springframework.boot.test.json.JacksonTester;
+import uk.gov.hmcts.reform.wataskmanagementapi.exceptions.BadRequestException;
+
+import java.io.IOException;
+
+import static java.util.List.of;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.SearchOperator.BOOLEAN;
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.SearchOperator.IN;
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.parameter.SearchParameterKey.AVAILABLE_TASKS_ONLY;
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.search.parameter.SearchParameterKey.JURISDICTION;
+
+@JsonTest
+@ExtendWith(MockitoExtension.class)
+class SearchRequestCustomDeserializerTest {
+
+    @Autowired
+    private JacksonTester<SearchParameter<?>> json;
+
+
+    @Mock
+    private JsonParser jsonParser;
+
+    @Mock
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private JsonNode searchNode;
+    @Mock
+    private JsonNode operatorNode;
+
+    @Mock
+    private DeserializationContext deserializationContext;
+
+    @Test
+    void testSearchParameterBooleanOperatorIsDeserialized() throws IOException {
```

Comment:
try adding new test names as `snake_case`.  I know there are a mix already.

Reply:
This class has nothing to do with snake case. But, I have renamed the tests to make it more sense.